artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQServerLogger.java

```diff
@@ -35,30 +32,26 @@
 import org.apache.activemq.artemis.core.config.Configuration;
 import org.apache.activemq.artemis.core.io.IOCallback;
 import org.apache.activemq.artemis.core.io.SequentialFile;
-import org.apache.activemq.artemis.core.paging.cursor.PagePosition;
-import org.apache.activemq.artemis.core.paging.cursor.PageSubscription;
-import org.apache.activemq.artemis.core.persistence.OperationContext;
 import org.apache.activemq.artemis.core.protocol.core.Packet;
 import org.apache.activemq.artemis.core.protocol.core.impl.wireformat.BackupReplicationStartFailedMessage;
 import org.apache.activemq.artemis.core.remoting.impl.netty.TransportConstants;
-import org.apache.activemq.artemis.core.server.routing.targets.Target;
 import org.apache.activemq.artemis.core.server.cluster.Bridge;
 import org.apache.activemq.artemis.core.server.cluster.impl.BridgeImpl;
 import org.apache.activemq.artemis.core.server.cluster.impl.ClusterConnectionImpl;
 import org.apache.activemq.artemis.core.server.cluster.quorum.ServerConnectVote;
 import org.apache.activemq.artemis.core.server.impl.ActiveMQServerImpl;
 import org.apache.activemq.artemis.core.server.impl.ServerSessionImpl;
 import org.apache.activemq.artemis.core.server.management.Notification;
+import org.apache.activemq.artemis.core.server.routing.targets.Target;
+import org.apache.activemq.artemis.logs.BundleFactory;
 import org.apache.activemq.artemis.logs.annotation.LogBundle;
 import org.apache.activemq.artemis.logs.annotation.LogMessage;
-import org.apache.activemq.artemis.logs.BundleFactory;
 import org.apache.activemq.artemis.spi.core.remoting.Connection;
-import org.w3c.dom.Node;
 
 /**
  * Logger Code 22
  */
-@LogBundle(projectCode = "AMQ", regexID = "22[0-9]{4}")
+@LogBundle(projectCode = "AMQ", regexID = "22[0-9]{4}", retiredIDs = "221026, 221052, 222012, 222020, 222021, 222022, 222024, 222027, 222028, 222029, 222052, 222071, 222078, 222079, 222083, 222084, 222088, 222090, 222102, 222128, 222134, 222135, 222152, 222159, 222163, 222167, 222170, 222171, 222182, 222192, 222193, 222204, 222252, 222255, 222257, 224001, 224002, 224003, 224005, 224035, 224100, 224121, 224127, 222260, 222259, 222288")
```

Comment:
It is a bit ugly, but I'm not sure there are other viable options given this has to operate in the confines of an [annotation processor](https://docs.oracle.com/javase/8/docs/api/javax/annotation/processing/Processor.html). I'm no expert here so I could be wrong.

Reply:
Regardless whether we separated it out, its not clear that there is a benefit to using String rather than e.g. just int[] ?

If trying to split it out, it would just be another annotation to process. Processors can handle any they like really. In this case it handles one main annotation, which it looks up at the start of the processing for each of the indicated interface/class that were found marked (and then in processing each, further handles multiple other related annotations found on methods). It could just look up another annotation from the class/interface. Or Many, if multiples were allowed (might help combat ultra-long lists getting too ugly).

Reply:
How about something like this?
```java
@LogBundle(
   projectCode = "AMQ",
   regexID = "22[0-9]{4}",
   retiredIDs = {
      221026,
      221052,
      222012,
      222020,
      222021,
      222022,
      222024,
      222027,
      222028,
      222029,
      222052,
      222071,
      222078,
      222079,
      222083,
      222084,
      222088,
      222090,
      222102,
      222128,
      222134,
      222135,
      222152,
      222159,
      222163,
      222167,
      222170,
      222171,
      222182,
      222192,
      222193,
      222204,
      222252,
      222255,
      222257,
      224001,
      224002,
      224003,
      224005,
      224035,
      224100,
      224121,
      224127,
      222260,
      222259,
      222288})
```

Reply:
Think the int[] would be good simplify things in the impl.

Prefer the ugly single-line than a monitor-height list with individual code per line though hehe.

Reply:
Agreed.